Sources/SwiftDocC/Servers/FileServer.swift

```diff
@@ -56,49 +54,49 @@ public class FileServer {
     /**
      Returns the data for a given URL.
      */
-    public func data(for url: URL) -> Data? {
+    public func data(for path: String) -> Data? {
         let providerKey = providers.keys.sorted { (l, r) -> Bool in
             l.count > r.count
-            }.filter { (path) -> Bool in
-                return url.path.trimmingCharacters(in: slashCharSet).hasPrefix(path)
+            }.filter { (providerPath) -> Bool in
+                return path.trimmingCharacters(in: slashCharSet).hasPrefix(providerPath)
             }.first ?? "" //in case missing an exact match, get the root one
         guard let provider = providers[providerKey] else {
             fatalError("A provider has not been passed to a FileServer.")
         }
-        return provider.data(for: url.path.trimmingCharacters(in: slashCharSet).removingPrefix(providerKey))
+        return provider.data(for: path.trimmingCharacters(in: slashCharSet).removingPrefix(providerKey))
     }
     
     /**
      Returns a tuple with a response and the given data.
      - Parameter request: The request coming from a web client.
      - Returns: The response and data which are going to be served to the client.
      */
-    public func response(to request: URLRequest) -> (URLResponse, Data?) {
-        guard let url = request.url else {
-            return (HTTPURLResponse(url: baseURL, statusCode: 400, httpVersion: "HTTP/1.1", headerFields: nil)!, nil)
+    public func response(to request: HTTPRequest) -> (HTTPTypes.HTTPResponse, Data?) {
```

Comment:
This is a source breaking change. Is there anywhere to maintain both declarations to ease this transition?

Reply:
I can guard the old function that uses `URLRequest` and `URLResponse` on `#if canImport(Darwin)` if that's ok? Otherwise its presence on non-Darwin platforms adds a dependency `FoundationNetworking` with curl dependencies, which is what this PR was meant to remove in the first place.

Reply:
I'm not sure. I'm concerned that someone could have an existing setup on Linux with curl and the rest of the system networking libraries and that their setup would break when they update to a new DocC version. 

We try to provide a transition period for breaking changes. If that's not possible then so be it (although we would try to inform people about it ahead of time in the Forums). But it there's a way to stage this, then I think that it would be less disruptive that way.

Would it be possible to use a compile time define to exclude FoundationNetworking? That way we could have a 3 stage transition (first opt-in to exclude, then exclude by default, and finally remove it completely)